integration-tests-support/test-support/src/main/java/org/apache/camel/quarkus/test/AvailablePortFinder.java

```diff
@@ -54,15 +57,23 @@ private AvailablePortFinder() {
      * @return                       the available port
      */
     public static int getNextAvailable() {
+        // Using AvailablePortFinder in native applications can be problematic
+        // E.g The reserved port may be allocated at build time and preserved indefinitely at runtime. I.e it never changes on each execution of the native application
+        logWarningIfNativeApplication();
+
         while (true) {
             try (ServerSocket ss = new ServerSocket()) {
                 ss.setReuseAddress(true);
                 ss.bind(new InetSocketAddress((InetAddress) null, 0), 1);
 
                 int port = ss.getLocalPort();
                 if (!isQuarkusReservedPort(port)) {
-                    LOGGER.info("getNextAvailable() -> {}", port);
-                    return port;
+                    String callerClassName = getCallerClassName();
+                    String value = RESERVED_PORTS.putIfAbsent(port, callerClassName);
```

Comment:
Is this making some assumptions about whether this class can concurrently exist in multiple JVMs or multiple class loaders within a single maven invocation? IIRC surefire is forking a new JVM for each Maven module. So if building with mvnd, the AvailablePortFinder class may exist in several concurrent JVMs.

Reply:
Can you elaborate a bit more? It's not entirely clear to me why it matters in this context.

All it's doing is trying to track which class reserved a specific port. Arguably we don't need to do that. It just provides some nice logging output and a way to clear out stale entries from the `RESERVED_PORTS` map.

Reply:
Thanks for explaining. If it is just for logging, indeed, multiple RESERVED_PORTS instances in concurrent per-module JVMs do not matter.